src/lib.rs

```diff
@@ -58,26 +59,26 @@ impl HyperLogLog {
         HyperLogLog {
             alpha: hll.alpha,
             p: hll.p,
-            m: hll.m,
-            M: vec![0; hll.m],
+            number_of_registers: hll.number_of_registers,
+            registers: vec![0; hll.number_of_registers],
             sip: hll.sip,
         }
     }
 
     /// Insert a new value into the `HyperLogLog` counter.
     pub fn insert<V: Hash>(&mut self, value: &V) {
-        let sip = &mut self.sip.clone();
-        value.hash(sip);
+        let mut sip = self.sip;
+        value.hash(&mut sip);
         let x = sip.finish();
```

Comment:
Are you sure this is correct? I'm under the impression that `SipHasher13` maintains state that isn't cleared by `finish`, meaning that future identical `value`'s will hash differently.

Reply:
I think let mut sip = self.sip; implicitly clones the value.

Reply:
Ah, right, it implements `Copy`.

Reply:
Yeah, although I think replacing it with a generic that implements the Hasher trait would be cleaner.